.travis.yml

```diff
@@ -77,13 +61,17 @@ addons:
     build_command: "travis_wait mvn clean package -B -DskipTests=true"
     branch_pattern: coverity_scan
 
+install:
+  # Build and install current module and dependencies. Handle output timeouts and retry on error.
+  - travis_retry travis_wait mvn install -DskipTests=true -Dmaven.javadoc.skip=true -B -V -pl janusgraph-${MODULE} -am
+
 script:
   - if [[ "${COVERITY_SCAN_BRANCH}}" == 1 ]]; then
       echo "Skipping build/test on 'coverity_scan' branch.";
       exit 0;
     fi
-  - mvn install -DskipTests=true -B -pl janusgraph-${MODULE} -am
-  - mvn verify -pl janusgraph-${MODULE} ${ARGS}
+  # Build and test module. Handle output timeouts and retry with a clean build on error.
+  - travis_retry travis_wait 50 mvn clean verify -pl janusgraph-${MODULE} ${ARGS}
```

Comment:
It will only try 3 times (unfortunately) or up to a total of 50 minutes, whichever comes first. It retries the whole given command so it'll do a clean build and execute all relevant tests on each retry.